ractor.c

```diff
@@ -1932,6 +1932,100 @@ ractor_moved_missing(int argc, VALUE *argv, VALUE self)
     rb_raise(rb_eRactorMovedError, "can not send any methods to a moved object");
 }
 
+/*
+ *  Document-class: Ractor::ClosedError
+ *
+ *  Raised when an attempt is made to take something from the Ractor's outgoing port,
+ *  but it is closed with Ractor#close_outgoing, or to send something to Ractor's
+ *  incoming port, and it is closed with Ractor#close_incoming, or an attempt to
+ *  send/take something with ractor which was already terminated.
```

Comment:
Not 100% accurate, as you can still read from a non-empty Queue after it has been closed. Also, you can't push to a closed outgoing port, nor read from a non-empty incoming port.  I'd rephrase:

Raised when an attempt is made to send a message to a closed port, or to retrieve a message from a closed and empty port. Ports may be closed explicitly with  Ractor#close_outgoing/close_incoming and are closed implicitly when a Ractor terminates.